io_port.c

```diff
@@ -105,30 +77,127 @@ io_port_deregister(void)
                io_port.addr + io_port.size - 1);
 
     io_port.addr = IO_PORT_UNMAPPED;
+
+    free(io_port.readl);
+    free(io_port.writel);
 }
 
 void
 io_port_write(uint64_t addr, uint64_t size, uint32_t val)
 {
+    uint64_t port_index;
+    uint64_t alignment;
+
     assert(io_port.enable && io_port.addr <= addr
            && addr < (io_port.addr + io_port.size));
 
     addr -= io_port.addr;
+    alignment = addr % sizeof(uint32_t);
+
+    if (size == 0) {
+       DBG("Zero sized write to %" PRIu64 "\n", addr);
```

Comment:
This should probably return here to avoid the callbacks needing to handle weird 0 size writes.

Reply:
Oh, yeah, i ment to put that there.